src/thunderbots/software/test/geom/util.cpp

```diff
@@ -0,0 +1,794 @@
+#include "geom/util.h"
+#include <gtest/gtest.h>
+#include <cmath>
+#include <cstdlib>
+#include <ctime>
+#include <iostream>
+#include <sstream>
+#include "geom/angle.h"
+#include "geom/point.h"
+
+// Set this to 1 to enable debug output.
+#define DEBUG 0
+
+#if DEBUG
+#define dbgout std::cout
+#else
+std::ostringstream dbgout;
+#endif
+
+TEST(GeomUtilTest, dist_line_vector2)
+{
+    double calculated_val, expected_val;
+
+    // case 1
+    Line test1line(Point(0, 1), Point(0, 0));
+    Point test1p(2, 0);
+    calculated_val = dist(test1line, test1p);
+    expected_val   = 2;
+    EXPECT_EQ(expected_val, calculated_val);
+
+    // case 2
+    Line test2line(Point(2, 0), Point(0, 2));
+    Point test2p(0, 0);
+    calculated_val = dist(test2line, test2p);
+    expected_val   = sqrt(2);
+    EXPECT_DOUBLE_EQ(expected_val, calculated_val);
+
+    // case 3
+    Line test3line(Point(0, 0), Point(0, 0));
+    Point test3p(1, 0);
+    calculated_val = dist(test3line, test3p);
+    expected_val   = 1;
+    EXPECT_EQ(expected_val, calculated_val);
+
+    Line line(Point(1, -1), Point(5, -2));
+    Point p(2, -3);
+
+    EXPECT_NEAR(1.69775, dist(line, p), 1e-5);
+
+    p = Point(2, 1);
+    EXPECT_NEAR(2.18282, dist(line, p), 1e-5);
+
+    p = Point(2, 0);
+    EXPECT_NEAR(1.21268, dist(line, p), 1e-5);
+}
+
+TEST(GeomUtilTest, test_proj_len)
+{
+    double calculated_val, expected_val;
+    // test case 1
+    Point test1p1(0, 0);
+    Point test1p2(4, 4);
+    Point test1p3(4, 0);
+    calculated_val = proj_len(Seg(test1p1, test1p2), test1p3);
+    expected_val   = 2 * sqrt(2);
+    EXPECT_DOUBLE_EQ(expected_val, calculated_val);
+
+    // test case 2
+    Point test2p1(0, 0);
+    Point test2p2(4, 0);
+    Point test2p3(4, 4);
+    calculated_val = proj_len(Seg(test2p1, test2p2), test2p3);
+    expected_val   = 4;
+    EXPECT_EQ(expected_val, calculated_val);
+
+    // test case 3
+    Point test3p1(0, 0);
+    Point test3p2(4, 4);
+    Point test3p3(-4, -4);
+    calculated_val = proj_len(Seg(test3p1, test3p2), test3p3);
+    expected_val   = -4 * sqrt(2);
+    EXPECT_DOUBLE_EQ(expected_val, calculated_val);
+
+    // test case 4
+    Point test4p1(0, 0);
+    Point test4p2(4, 1);
+    Point test4p3(-4, -4);
+    calculated_val = proj_len(Seg(test4p1, test4p2), test4p3);
+    expected_val   = -sqrt(32) * (cos((M_PI / 4.0f) - atan(1.0f / 4.0f)));
+    EXPECT_DOUBLE_EQ(expected_val, calculated_val);
+}
+
+TEST(GeomUtilTest, test_contains_triangle_point)
+{
+    // this triangle lies in the first quatren of the field, we can rota
+    Point p1(0, 0);
+    Point p2((std::rand() % 100) / 100, 0);
+    Point p3((std::rand() % 100) / 100, (std::rand() % 100) / 100);
+    Point p((std::rand() % 100) / 100, (std::rand() % 100) / 100);
+    bool expected_val = true;
+
+    /* i don't know what is going on here, this part seems to not work very well
```

Comment:
Dead code. Unless it's useful and should be uncommented, delete it.

Reply:
bump

Reply:
done